### PR TITLE
[13.0][MIG] web_switch_context_warning

### PR DESCRIPTION
**Pull request by JoanSForgeFlow**
**web_switch_context_warning** migration to v13:

In Odoo 13 company is a user's attribute, it's possible to work in different tabs with different companies. Because of this, now the module only shows a warning when user or database have been switched.

cc ~~ @LoisRForgeFlow @AdriaGForgeFlow

### CODE REVIEW

File: web_switch_context_warning/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Shows a warning if current user or database have been switched in another tab or window.
```
Comment (AdriaGForgeFlow) [inline web_switch_context_warning/readme/DESCRIPTION.rst]: Could you update this as well to add that user changes are also warned

File: web_switch_context_warning/readme/ROADMAP.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+    * If the browser don't implement Shared Worker (http://www.w3.org/TR/workers/#sharedworker), the warning message will not be displayed (there is no polyfill).
+
+    * Switching company in a separate browser or in private browsing mode will not be detected by this module. It's a limitation of Shared Worker(limit to browser session, server:port...)
```
Comment (AdriaGForgeFlow) [inline web_switch_context_warning/readme/ROADMAP.rst]: Could you update this, please.

File: web_switch_context_warning/static/src/js/switch_context_warning.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    var UserMenu = require("web.UserMenu");
+    var session = require("web.session");
+    // Show a big banner in the top of the page if the context has been
+    // changed in another tab or window (in the same browser)
+
+    if (!window.SharedWorker) {
+        // Not supported
+        return;
+    }
+    var SwitchCompanyWarningWidget = Widget.extend({
```
Comment (AdriaGForgeFlow) [inline web_switch_context_warning/static/src/js/switch_context_warning.js]: In this new version maybe changing this to `SwitchContextWarningWidget` makes more sense. What do you think?

File: web_switch_context_warning/static/src/js/switch_context_warning.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        },
+        generateSignature: function() {
+            return [session.uid, session.db].join();
+        },
+    });
+
+    UserMenu.include({
+        init: function(parent) {
+            this._super(parent);
+            var switchCompanyWarning = new SwitchCompanyWarningWidget();
```
Comment (AdriaGForgeFlow) [inline web_switch_context_warning/static/src/js/switch_context_warning.js]: Same here

File: web_switch_context_warning/static/src/js/switch_context_warning.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            switchCompanyWarning.insertAfter(".o_main_navbar");
+        },
+    });
+
+    return SwitchCompanyWarningWidget;
```
Comment (AdriaGForgeFlow) [inline web_switch_context_warning/static/src/js/switch_context_warning.js]: Same here

**Review by AdriaGForgeFlow — COMMENTED** [review]
Some comments below.

File: web_switch_context_warning/static/src/js/switch_context_warning.js (hunk 3 above)
Comment (JoanSForgeFlow) [reply web_switch_context_warning/static/src/js/switch_context_warning.js]: Of course. now I'm going to change it.

**Review by AdriaGForgeFlow — APPROVED** [review]
Functional and code review :+1:

**Review by LoisRForgeFlow — COMMENTED** [review]
Not working for me, could you have a look?

**Pull request closed without merging**